documents/description.md

```diff
@@ -2,8 +2,62 @@
+- Musical pieces of specific genres rely on a limited set of patterns and cliches.  Thus, instrumentalists will often find themselves playing the same phrases over and over when sight-reading "new" sheet music. This is good because it allows instrumentalists to quickly recognize common patterns, but it's bad because it's not entirely sight-reading. Practicing this way creates a sight-reading "crutch" that can't always be depended upon.  Who knows when a musician will have to sight-read in a drastically different genre than what they are used to?
+
+The goal of this DSL is to alleviate these issues by allowing users to generate an infinite amount of random sheet music based on highly-customizable parameters.
+
 ## Language domain
 
+The domain of this DSL is parameterized random (is there a word for this?) sheet music generation.  Instrumentalists or vocalists who wish to practice sight-reading/singing would benefit from this DSL.
+There exist a few programs/DSLs in this domain.  
+- [randomsheetmusic.com](http://www.randomsheetmusic.com/) generates random simple sheet music.  It is very easily accessible, but also quite limited.  It only generates single-clef, monophonic melodies and has only a few parameters that users can change.  The simple design (visually) is something I'm after, but I'd of course like to have more expanded functionality in my DSL.
+- [Melisma Stochastic Melody Generator](http://www.link.cs.cmu.edu/melody-generator/) gives a lot more control to users, but is also monophonic and doesn't generate viewable sheet music (only MIDI).  Some of the parameter names are also a bit confusing (rhythmic anchoring, key profile type).  I'd take some cues from some of the parameters, but I'd strive to make them easier to understand for beginner instrumentalists and for young kids (11 to 12-year-olds, maybe).
+- [Impro-visor](https://www.cs.hmc.edu/~keller/jazz/improvisor/) is a flexible program that generates sheet music and MIDI of single-line  improvisations over given chord changes. This program has a level of refinement and sophistication that I hope to achieve, but its focus is slightly different than mine.  It's meant to help with improvising rather than sight-reading, so its more geared toward generating licks than generating customizable sheet music.
+- [Sight Reading Factory](https://www.sightreadingfactory.com/) is a paid software that generates random sheet music for various instruments and difficulty levels.  The presentation is great and the underlying generation algorithms work really well.  However, it doesn't provide as much flexibility to users as I'd like my DSL to (for example, it doesn't support key modulation, key signature changes, arbitrary numbers of staves, detailed customization of chords or rhythm, etc).
+
+Note:  All of the above are GUI-based, while my DSL won't be (at least in its first iteration).
```

Comment:
Here is the paper on the grammar (useful info starts in part 6, and continues until constraints): http://smc07.uoa.gr/SMC07%20Proceedings/SMC07%20Paper%2055.pdf 

Some of the parts of the DSL are a lot more specific than what you would do, but I think the way the program categorizes notes, then allows for markov type chains might help in giving you ideas for how to form your DSL. So even if you just specified maybe that no 'bad' notes (out of chord) are allowed, something like this might be helpful. It also goes into specifying minimum and maximum values for rhythm, intervals between notes. You might not want the chain, but the types of things that are specified in this grammar might help show you how similar things were represented in an already developed textual grammar.